local_peer.go

```diff
@@ -8,13 +8,20 @@ import (
 	"time"
 )
 
+const (
+	deferTopologyUpdateDuration = 1 * time.Second
+)
+
 // localPeer is the only "active" peer in the mesh. It extends Peer with
 // additional behaviors, mostly to retrieve and manage connection state.
 type localPeer struct {
 	sync.RWMutex
 	*Peer
-	router     *Router
-	actionChan chan<- localPeerAction
+	router                *Router
+	actionChan            chan<- localPeerAction
+	topologyUpdates       [][]*Peer
```

Comment:
Given that we're eventually going to build a set of peer names from this data, why not hold a `peerNameSet` here?